components/camel-qdrant/src/main/java/org/apache/camel/component/qdrant/QdrantProducer.java

```diff
@@ -0,0 +1,264 @@
+                                Points.ReadConsistency.class)),
+                (r, t) -> {
+                    if (t != null) {
+                        exchange.setException(new QdrantActionException(QdrantAction.DELETE, t));
+                    } else {
+                        in.setBody(new ArrayList<>(r));
+                        in.setHeader(Qdrant.Headers.SIZE, r.size());
+                    }
+
+                    callback.done(false);
+                });
+
+        return false;
+    }
+
+    private boolean delete(Exchange exchange, AsyncCallback callback) {
+        final String collection = getEndpoint().getCollection();
+        final Message in = exchange.getMessage();
+        final Object body = in.getBody();
+
+        Points.PointsSelector selector;
+
+        if (body instanceof Points.PointsSelector) {
+            selector = (Points.PointsSelector) body;
+        } else if (body instanceof Points.PointId) {
+            selector = Points.PointsSelector.newBuilder()
+                    .setPoints(
+                            Points.PointsIdsList.newBuilder()
+                                    .addIds((Points.PointId) body)
+                                    .build())
+                    .build();
+        } else if (body instanceof Points.Filter) {
+            selector = Points.PointsSelector.newBuilder()
+                    .setFilter((Points.Filter) body)
+                    .build();
+        } else {
+            exchange.setException(new QdrantActionException(
+                    QdrantAction.DELETE,
+                    "A payload of type PointsSelector or Filter is expected"));
+
+            return true;
+        }
+
+        Points.DeletePoints value = Points.DeletePoints.newBuilder()
+                .setCollectionName(collection)
+                .setPoints(selector)
+                .setWait(true)
+                .build();
+
+        call(
+                this.client.deleteAsync(value),
+                (r, t) -> {
+                    if (t != null) {
+                        exchange.setException(new QdrantActionException(QdrantAction.DELETE, t));
+                    } else {
+                        in.setHeader(Qdrant.Headers.OPERATION_ID, r.getOperationId());
+                        in.setHeader(Qdrant.Headers.OPERATION_STATUS, r.getStatus().name());
+                        in.setHeader(Qdrant.Headers.OPERATION_STATUS_VALUE, r.getStatus().getNumber());
+                    }
+
+                    callback.done(false);
+                });
+
+        return false;
+    }
+
+    // ***************************************
+    //
+    // Helpers
+    //
+    // ***************************************
+
+    private CamelContext getCamelContext() {
+        return getEndpoint().getCamelContext();
+    }
+
+    private <T> void call(ListenableFuture<T> future, BiConsumer<T, Throwable> consumer) {
+        Futures.addCallback(
```

Comment:
One thing I've been thinking is whether we could avoid using Guava. It's always a source of alignment problems and whatnot. It would be great if we could minimize it (IMHO).

Reply:
They're are improving from this point of view. The releases are much more stable now.

Reply:
> They're are improving from this point of view. The releases are much more stable now.

Oh, that's great to know. I have acquired a bad impression of it because of the problems of the past, but I happy to know it's not problematic anymore (as I do recognize it's useful). In this case, I think it's safe to ignore my suggestion above.

Reply:
No we can't as that is a fundamental part of grpc-java i.e. the `ListenableFuture` is what the various methods of the client to return ... I know it is a pain but there is not much we can do as long a grpc-java is involved

Reply:
Marking this as resolved, as @oscerd clarified this is not a problem anymore. Thanks!